### DISCUSSION TITLE
Help Setup Fuji PXF4

### DISCUSSION THREAD

**Discussion by Dgocof** [artisan-roaster-scope/artisan#1086] [Q&A]
My Fuji PXF4 Setup. SEE PDF Below for screen shots.

[Fuji.Setup.pdf](https://github.com/artisan-roaster-scope/artisan/files/10541130/Fuji.Setup.pdf)

On Mac with Artisan 2.8.2
We are roasting on a Probat L25 with a maxtrol modulating gas valve. The signal is setup for current as the maxtrol is set for 4-20 Ma
Phidget TMP 1101 4 Thermocouple. 2nd BT Thermocouple and 3rd Thermocouple is environment Temp
I have 2 bean probes. 1 is 1mm and 2nd is 3 mm.
3 mm is hooked to BT from Fuji. Main Thermocouple.

My Problem: See PDF below.
This is my setup but I am not getting any control of the gas valve or flame either manual with a slider or with follow background.

I am mostly following the setup: https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-using-bt-as-input-fuji-pxf
Regarding the setup of the Fuji. Anyone who has got this working have you changed any of the default settings in the Fuji and why?
Is using the scan feature the way to verify that you are communicating completely with the Fuji?

I don’t seem to be communicating with the Modbus registers to operate the gas valve?
Regarding the setup of the Fuji. Anyone who has got this working have you changed any of the default settings in the Fuji and why?

With PID on, in Control, it doesn’t seem to matter.
I think the computer is communicating with the Fuji as I get BT. When I use the scan in port configuration I see no numbers coming up.

More Questions: See PDF Below
I’m curious what the difference is between Modbus and Modbus 34 in extra devices. Are these the 4 modbus channels?

I am also wondering if I choose PID (device Assignment) and Fuji. Can I control the gas valve with BT or only ET? Is the solution to this swapping BT/ET??

I’m also a bit confused with choosing meter (Modbus) in device assignment is that associated with serial and input 1-8? Top half of diagram below (Port Configuration)? See PDF

If I choose PID in device assignment that is referencing PID,in (port configuration) slave,PID. Button half of diagram below? SEE PDF

My setup is in this pdf. I wasn't able to post my screen shots here in this screen.

[Fuji.Setup.pdf](https://github.com/artisan-roaster-scope/artisan/files/10541130/Fuji.Setup.pdf)

**Comment by bigl00z3** [comment]
As you have also opened the topic here, I will also paste my answer here which I originally posted in the [issue post](https://github.com/artisan-roaster-scope/artisan/issues/1085):

Nico

--------

Dgocof,

this is most likely not an artisan issue but more a configuration issue. You should post this in the [Discussion](https://github.com/artisan-roaster-scope/artisan/discussions) page.

> I think the computer is communicating with the Fuji as I get BT. When I use the scan in port configuration I see no numbers coming up.

If you see the ports being listed during a scan, you are communicating with the Fuji.

> I’m curious what the difference is between Modbus and Modbus 34 in extra devices. Are these the 4 modbus channels?

Modbus is for inputs 1 and 2 in the ports window while Modbus 34 is for inputs 3 and 4. Each input is talking to one of the register on the Fuji (either to read or send a value). For example 2000 is the the PV value measure by your PID.  240 is the SV value (from PID pallet 1).
When you call for Modbus in the `Device Assignment/Meter,` it will look for Modbus inputs 1 and 2.

> My setup is in this pdf. I wasn't able to post my screen shots here in this screen.

it looks like your SV BT is not the wrong register. Your Modbus 3 is set to register is set to 2041 which is Duty%. SV BT is register 240. In your screenshot the value is -5 which is the minimum value sent. It can send from -5% to 105%
<img width="724" alt="image" src="https://user-images.githubusercontent.com/8886922/215689700-604068cf-49a3-42ed-bff9-ebe19d9fd5d1.png">

For the list of registers, you can check the [Fuji communication document](https://felib.fujielectric.co.jp/download/ContentDownload.htm?dataid=16281277&site=global&lang=en)


You can also check my 1st post when I started using the PID [Artisan – Roasting with the PID – Fuji PXF (and test of MaxWell MTA)](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-fuji-pxf-and-test-of-maxwell-mta) they are some additional information to the post you have been following.

> I am also wondering if I choose PID (device Assigment) and Fuji. Can I control the gas valve with BT or only ET?

Initially I used to control via ET and then moved to controlling from BT. Both are possible but as the processing is done by the Fuji itself, you need to connect (physically) the right probe to it (either ET or BT). If you want to control with the other probe you have two options:

- Add a second Fuji PXF (it will need to be slave 2), the 1st one connected to ET and the second one connected to BT. In the same way I had done on the above screenshot with the MaxWell on input n°2.
- When you want to change probe without adding a second Fuji nor rewiring, you could use the Artisan internal PID (slower). For this, change slave number from one to zero.


Nico

**Reply by Dgocof** [reply]
Hi Nico,
Still struggling. Could you take a look at my above post with screen shots and give me some guidance? I have 2 posts going.
it looks like your SV BT is not the wrong register. You told me this but to clarify, I do have the correct register?
Thanks

**Comment by Dgocof** [comment]
Nico did you change some settings in the Fuji?
Such as SV to SV1? 
What else beside Baud and parity?

**Reply by bigl00z3** [reply]
Sorry for the late reply. 
Yes, I am using SV1.  when looking at the PXF Modbus communication manual SV is local only and to be remote controlled you have to use one of the other SV. in my case I chose SV1:
<img width="955" alt="image" src="https://user-images.githubusercontent.com/8886922/218324796-05de3aa9-f4b5-45d6-9e6b-88d876238d5e.png">

Regarding baud rate and parity, I have it set to 19200:
<img width="272" alt="image" src="https://user-images.githubusercontent.com/8886922/218324909-32dec513-dd79-45cb-97d8-e3766ef6e9d9.png">
What is important is that you have to make sure that your PXF is set exactly to the same communication settings as in artisan.

I hope this helps.

Nico

**Comment by Dgocof** [comment]
When I changed the Fuji to SV1 I got the burners to respond finally but now the SV temp in Artisan is not the same as the SV temp shown on the Fuji. Unfortunately the temp only goes up to the SV temp indicated on the Fuji which is lower that the SV temp I set in Artisan. The higher I set the SV in Artisan the more the SV temp on the Fuji is lower. Where should I look to solve this? 
I was not able to find the setting for PID1 in the Fuji. Does this need to be need addressed also? Thanks

**Reply by bigl00z3** [reply]
The PID settings for SV1 are in CH2 menu of the Fuji PXF:
<img width="651" alt="image" src="https://user-images.githubusercontent.com/8886922/218521271-07d0a52d-9183-46f9-9127-f43cfc3a2343.png">
(Screenshot from the [PXF user manual](https://www.fujielectric.com/products/instruments/products/controller/PXF.html) ) 
You also need to check that the settings are correct in `CH6 SET (Setup Parameters) `of the Fuji.

Did you change anything in `CH1 PID (control parameters)` ? For example lines 59 and 60 in the below screenshot?
<img width="939" alt="image" src="https://user-images.githubusercontent.com/8886922/218524882-ea71ba69-3238-41a2-a21b-f4fc88430802.png">
There is the following note at the bottom of the page:
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/8886922/218525182-184ed800-d1c5-4c89-8717-56665ab4f762.png">

I am not at home right now and cannot check directly on my Fuji. But Maybe you can upload here your current Artisan-settings and I can have a look if I see anything on this side.

just in case, I will add below a screenshot of my `Config/Port...  Modbus settings ` What do have under the PID part?
<img width="1069" alt="image" src="https://user-images.githubusercontent.com/8886922/218523943-acd88807-661c-412c-907d-e1a6bffe0e74.png">
For mine I have SV factor of 10.

Nico

**Comment by Dgocof** [comment]
![image](https://user-images.githubusercontent.com/117209911/218566451-7e4e2e3c-878d-4f72-85dc-d60bdc89b6e1.png)
<img width="1300" alt="image" src="https://user-images.githubusercontent.com/117209911/218566586-2023e0fb-76fb-4296-96d6-63ca689afe2c.png">
<img width="1126" alt="image" src="https://user-images.githubusercontent.com/117209911/218566678-b76d8121-9dfc-49e2-ae31-73c08584a6a7.png">

I verified my Communication settings in Fuji to match Artisan. Baud, Parity etc...
I checked SVL it is default 32 and SVH is default 752
I'm addressing SV1 and communicating with the modbus.

 I can change SV in Artisan and the Fuji controller SV changes but it doesn't reflect the number in Artisan. For instance if I change the SV by the slider to 400 F, the Fuji SV will change to 325F. The higher the slider the more the SV temp on the Fuji is off. 
This is my last bug I cannot solve. 

I cannot find or access NO 8 (PID 1,2,3,etc..) in operation control parameter. Even when I go to all the hidden settings I can't find that one. I was looking to see if I could change that from LOCL to PID 1. LOCL is the default. 

Another thing when I use Ulnt16 for 2041 it I get 4 digits for the % number. In github I have read that in the older version unlt16 has changed to snlt16 in the new version of software.??

I'm also wondering why we are addressing 00F7h (247) instead of 00f3h (243)? 
247 is Output convergence value and 243 is the D for PID . Just curious.

**Reply by bigl00z3** [reply]
On the  Fuji you need to have SV1 selected instead of LoCl
<img width="1183" alt="image" src="https://user-images.githubusercontent.com/8886922/218789129-031d61c2-72f0-4286-9b5f-fe850ed186c8.png">
Then your PID parameters will not be from `Ch1 PID (Control Parameters)` but from `CH2 PLT (PID palette parameters)` 

> I cannot find or access NO 8 (PID 1,2,3,etc..) in operation control parameter. 
I don't understand what you mean by this.

> I can change SV in Artisan and the Fuji controller SV changes but it doesn't reflect the number in Artisan. For instance if I change the SV by the slider to 400 F, the Fuji SV will change to 325F. The higher the slider the more the SV temp on the Fuji is off.

This rings a bell.  I don't have the Fuji with me to test. But I had a similar issue in the beginning when I wrongly set the Fuji's temperature range. If I remember correctly it was that I changed the % of the range. Looking at the manual it could be (but tbc) in the following settings. 
<img width="760" alt="image" src="https://user-images.githubusercontent.com/8886922/218795058-2294f5dd-d21a-4b1a-a0db-acf52d79504f.png">

or maybe also in:
<img width="691" alt="image" src="https://user-images.githubusercontent.com/8886922/218795176-214390e7-f954-4784-b0e4-48c70e4513a8.png">

Check the values of your PID versus the default values in the user manual.

> I'm also wondering why we are addressing 00F7h (247) instead of 00f3h (243)?
247 is Output convergence value and 243 is the D for PID . Just curious.

That's a good question and by looking at the manual it may be 243. I will need to check when I get back. When looking at the manual it is:
<img width="635" alt="image" src="https://user-images.githubusercontent.com/8886922/218798159-3d013e03-7d9a-425c-bdbb-e92ca7e9308f.png">
You can always try yourself. if you change the register for the "D", and you change it in Artisan/Control
<img width="176" alt="image" src="https://user-images.githubusercontent.com/8886922/218799387-20685d50-f257-48a5-967e-f294fd52f28f.png">
Try changing the D value, then press set:
<img width="492" alt="image" src="https://user-images.githubusercontent.com/8886922/218799718-bf655c21-1958-4615-82a0-6ddcec6d05fe.png">
Then have a look in the Fuji PXF's menu to see if the value has changed. If it didn't, it's probably that I made a mistake in my screenshot. As I haven't changed those values for so long (it just works perfectly like this...) I can't quite remember.  

Hopefully this time it will help you solve it.

Nico

**Reply by bigl00z3** [reply]
> _**I'm also wondering why we are addressing 00F7h (247) instead of 00f3h (243)?
> 247 is Output convergence value and 243 is the D for PID . Just curious.**_
> 
> That's a good question and by looking at the manual it may be 243.  I will need to check when I get back. 

I just had a look, and you were right, The D value is on register 00f3h (243) and not 00F7h (247) . I need to correct the screenshot on my website.

Nico

**Comment by Dgocof** [comment]
So along with all the settings in Artisan I have changed some settings in the Fuji. I'm not sure if they are correct yet.
 I am still having problems with following the back round BT temp I made in designer. Below is a screen print of how far off I am. The Blue line is my BT.

In the Fuji I have changed in the control parameters:
Operation control parameters:
3 REM  from Locl to Rem to look for the changes in Artisan. Not sure if this is necessary.
7 SVN from Locl to SV1
Chan 1 Set
59 is 0
60 is 999.9
Ch 6 Set
530 to J3
531 to 0
532 to 999.9  
533 to .1 for digit which changes digits in PID settings also. so my understanding by looking at the PID settings is if you have one digit, your PID setting in Artisan at .3 will actually be 3 in the Fuji. At least that is how mine reads.
Chan 9 Com
761 is 1
762 is 19200
763 is N
764 is 1.0      I'm not  sure which setting this matches in Chan 9

<img width="1345" alt="image" src="https://user-images.githubusercontent.com/117209911/222237555-ad3cb48c-9bc0-4241-9695-511ac64948e2.png">

**Reply by bigl00z3** [reply]
> So along with all the settings in Artisan I have changed some settings in the Fuji. I'm not sure if they are correct yet.

If you are not sure if the parameter is correct and that it is not comunicating, try changing the value of the Register in Artisan  and then check in the PID menu to see if the value has changed. If it didn't you probably have the wrong value, if it did, you have the right value.

As an  example (to check if you have the correct value or address) by creating buttons to send a value to the specific register by using a Modbus command such as for SV register n°240 (or any other register) `write(1,240,1)` where `write(slaveId,register,value)` .

Once you are sure that Artisan and the Fuji PXF are communicating well, you can start to work on the PID settings (the actual temperature regulation) to have BT follow as well as possible your BG.

Adjusting PID is pretty much an art and the best is to read one of the many explanation on how to hadapt it.

On my side, the approach I took, and which I describe on [my blog](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-using-bt-as-input-fuji-pxf), was to use the Fuji PXF's internal "Auto Tune" function which on my roaster found very good settings. But keep in mind that my roaster is a hot air / fluid bed roaster with low thermal inertia. 
In my case it worked well but on other roaster it may need more fine tuning.

The AutoTune is in :
<img width="954" alt="image" src="https://user-images.githubusercontent.com/8886922/222893591-88005b40-b1eb-4804-8f47-e20c9ffa00e6.png">

The best way to use the autotune is to preheat correctly your roaster. 
Then stabilise it (with beans) to a static temperature. In [my case](https://www.bigloose.com/v11/images/stories/blog/220105_PID_BT/Picture_9.png), I did it at 160°C. 
Once you decide it is stable enough, you can start the "auto tune" function from your PCF's front panel menu. It will start testing your system with heater waves to analyse which way it thinks the configuration is at it's best. 

When done, the "AT" display will disapear from the PXF's LCD.  and the temperature should be stable again.

At that point you can test how well (or not) it is following the curve.

If the autotune function is sufficient, you have to then go the long and complicated process of manually setting the P, I and D values. 

Nico
